.github/workflows/ci.yml

```diff
@@ -17,16 +17,36 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 10
 
+  cache-dependencies:
+    name: Cache dependencies
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: pnpm/action-setup@v4
+      - name: Install Node.js
+        uses: actions/setup-node@v4
```

Comment:
Prefer to use [volta](https://volta.sh/) as in the other existing builds in this file:

```
      - name: Install Node
        uses: volta-cli/action@v4
```

This ensures the CI build uses the same version of node that developers would have running locally (if they have volta installed)